### PR TITLE
fix(enterprise): mark nodes from unhealthy coordinators as lost

### LINKED ISSUE #13041: Fix error in PGCoordinator handling of missed heartbeats
When a coordinator misses heartbeats we incorrectly send updates that the nodes are disconnected as opposed to lost

### PR DESCRIPTION
**Pull request by coadler**
Fixes https://github.com/coder/coder/issues/13041

Instead of removing the mappings of unhealthy coordinators entirely,
mark them as lost. This prevents peers from disappearing from
other peers if a coordinator misses a heartbeat.

### CODE REVIEW

**Comment by coadler** [comment]
* **#13123** <a href="https://app.graphite.dev/github/pr/coder/coder/13123?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @coadler and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by spikecurtis — COMMENTED** [review]
This LGTM, but I'm relying on reading code to convince myself that it solves the high level use case of sending LOST updates to peers when a coordinator fails heartbeats.

In addition to the unit test on the heartbeats subcomponent, I'd like to see a test in a similar vein to `TestPGCoordinatorSingle_MissedHeartbeats` where we simulate a second coordinator entirely by DB calls, make it miss its heartbeats, and then verify that the first coordinator sends out a LOST update.

File: enterprise/tailnet/pgcoord_test.go [hunk 1]
```diff
@@ -415,6 +416,55 @@
...
+	ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitSuperLong)
+	defer cancel()
+	logger := slogtest.Make(t, nil).Leveled(slog.LevelDebug)
+
+	coordinator, err := tailnet.NewPGCoord(ctx, logger, ps, store)
+	require.NoError(t, err)
+	defer coordinator.Close()
+
+	agent := test.NewPeer(ctx, t, coordinator, "agent")
+	defer agent.Close(ctx)
```
Comment (spikecurtis) [inline enterprise/tailnet/pgcoord_test.go:434]: I think the agent on the real coordinator is superfluous to this test.  Just need to test the agent update on the fake coordinator, then have it miss heartbeats.

**Review by spikecurtis — APPROVED** [review]
LGTM!  Minor suggestion inline, but I don't need to review again

File: enterprise/tailnet/pgcoord_test.go (hunk 1 above)
Comment (coadler) [reply enterprise/tailnet/pgcoord_test.go:434]: Ah, good point.

**Pull request merged**